src/coreclr/src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -14,6 +15,11 @@ internal static class StartupHookProvider
         private const string InitializeMethodName = "Initialize";
         private const string DisallowedSimpleAssemblyNameSuffix = ".dll";
 
+        private static bool IsSupported { get; } = StartupHookProviderIsSupported();
+
+        private static bool StartupHookProviderIsSupported() =>
+            AppContext.TryGetSwitch("System.StartupHookProvider.IsSupported", out bool isSupported) ? isSupported : true;
+
```

Comment:
```suggestion
        private static bool IsSupported => AppContext.TryGetSwitch("System.StartupHookProvider.IsSupported", out bool isSupported) ? isSupported : true;

```

This value is used exactly once during process launch. It is not necessary to cache it in a static.